src/amo/reducers/reviews.js

```diff
@@ -261,28 +261,74 @@ export const selectLatestUserReview = ({
   return selectReview(reviewsState, userReviewId);
 };
 
+export function isReviewUpdate({
+  reviewsState,
+  newReviewId,
+  newReviewBody,
+}: {|
+  reviewsState: ReviewsState,
+  newReviewId: number,
+  newReviewBody: ?string,
+|}) {
+  const existingReview = selectReview(reviewsState, newReviewId);
+  if (!existingReview) {
+    return false;
+  }
+
+  // If this update is actually turning a rating into a review then
+  // it's not an update.
+  if (!existingReview.body && newReviewBody) {
```

Comment:
This causes a problem for the User Profile page. If I click the "Write a review" button for a rating-only review, when I submit the review it is not treated as an update and therefore the page is refreshed. If I also had a text box open editing an existing review at the time, I would lose my state. 

I don't think this is that big a deal, as it's a more extreme edge case, and I'm not sure how we can make it work the way we want both from the User Profile page and from the RatingManager (because that *does* wanted it treated like an insert and not an update), but it's some food for thought.

Reply:
> [While submitting a new review, ] if I also had a text box open editing an existing review at the time, I would lose my state.

I don't think that's worth fixing. The user would not do this intentionally.

Reply:
> If I click the "Write a review" button for a rating-only review, when I submit the review it is not treated as an update and therefore the page is refreshed

I have a fix for this.